org.OpenT2T.Onboarding.LightifyHub/node/hub.js

```diff
@@ -0,0 +1,233 @@
+                    {
+                        deferred.resolve(body);
+                    }
+                    else
+                    {
+                        var parsedBody = JSON.parse(body);
+                        deferred.resolve(parsedBody);
+                    }
+                }
+            });
+
+            response.on('error', function(e) {
+                deferred.reject(e);
+            });
+        });
+
+        request.on('error', (e) => {
+            deferred.reject(e);
+        });
+
+        if (content)
+        {
+            request.write(content);
+        }
+
+        request.end();
+
+        return deferred.promise;
+    },
+};
+
+module.exports = 
+{
+    inputNeeded : [
+        {
+            type: 'input',
+            name: 'deviceSerialNumber',
+            message: 'Osram Lightify Device Serial Number (alpha-numeric without trailing -XX): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid Serial Number.';
+                }
+            }
+        },
+        {
+            type: 'input',
+            name: 'username',
+            message: 'Osram Lightify User Name (create this in the Osram Lightify app): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid User Name.';
+                }
+            }
+        },
+        {
+            type: 'password',
+            name: 'password',
+            message: 'Osram Lightify Password (create this in the Osram Lightify app): ',
+            validate: function(value) {
+                var pass = !!value;
+                if (pass) {
+                    return true;
+                } else {
+                    return 'Please enter a valid Password.';
+                }
+            }
+        }
+    ],
+
+    // devices list as returned by the server 
+    devices: undefined,
+
+    // the is the serial number of the Lightify hub device
+    deviceSerialNumber: undefined,
+
+    // this is the oauth security token for the current session
+    securityToken: undefined,
+
+    // authorizes to the server based on the properties saved in the object 
+    authorize: function(errorCallback) {
+        // get initial set of authorization info, including account server
+        return internal.getSecurityToken().then((data) => {
+            // get the list of devices
+            return internal.getDevices().then((devices) => {
+                this.securityToken = data.securityToken;
+                this.devices = devices;
+                this.deviceSerialNumber = internal.deviceSerialNumber;
+                return devices;
+            })
+        },
+        function (error) { 
+            throw error;
+        });
```

Comment:
You can omit this error function entirely since it doesn't do anything but rethrow. The returned promise will still fail with the error.